packages/block-editor/src/store/reducer.js

```diff
@@ -847,7 +863,12 @@ export const blocks = pipe(
 			case 'RECEIVE_BLOCKS':
 			case 'INSERT_BLOCKS': {
 				const newState = new Map( state );
-				getFlattenedBlockAttributes( action.blocks ).forEach(
+
+				// Adds the selected pattern category to the outer block.
+				const newBlocks =
+					addSelectedPatternCategoryToOuterBlock( action );
```

Comment:
This feels like it should be called conditionally.

Also it's feeling a little bit like a side effect in a reducer.

Are "Patterns" a concept in the block editor outside of the WordPress context? If not then should we be adding this to the block editor store?

Reply:
The original approach used a different reducer before this one, to try to keep the concerns separate. However this one addresses actions.block directly, so any changes we make to the blocks before this point are overridden by this one.